File: server/src/main/java/com/db/datastoreserver/domain/post/repository/PostRepository.java
package com.db.datastoreserver.domain.post.repository;

import com.db.datastoreserver.domain.post.Category;
import com.db.datastoreserver.domain.post.Post;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface PostRepository extends JpaRepository<Post, Long> {
    List<Post> findAllByCategory(Category category);
}

File: k-datastore-server/src/main/kotlin/com/db/kdatastoreserver/domain/repository/PostRepository.kt
package com.db.kdatastoreserver.domain.repository

import com.db.kdatastoreserver.domain.Category
import com.db.kdatastoreserver.domain.Post
import org.springframework.data.mongodb.repository.MongoRepository

interface PostRepository : MongoRepository<Post, String> {
    fun findAllByCategoryIn(categories: List<Category>): List<Post>
}